### DISCUSSION TITLE
Scalar bar postion control on client side

### DISCUSSION THREAD

**Discussion by Jimmy-KL** [Kitware/trame#426] [Q&A] [answered]
Before asking this question, I have checked this similar discussion [#157](https://github.com/Kitware/trame/discussions/157), but it doesn't solve my problem. 

Using remote rendering, I can control the position of the scalar bar, but using local rendering, it seems I cannot. Here is a simple example.

```
from trame.app import get_server
from trame.ui.vuetify3 import SinglePageLayout
from trame.widgets import vuetify3

import pyvista as pv
from pyvista.trame.ui import plotter_ui

# -----------------------------------------------------------------------------
# Trame initialization
# -----------------------------------------------------------------------------
pv.global_theme.allow_empty_mesh = True
pv.OFF_SCREEN = True

server = get_server()
state, ctrl = server.state, server.controller

# -----------------------------------------------------------------------------
# Pipelines
# -----------------------------------------------------------------------------

mesh = pv.Sphere()
mesh["data"] = mesh.points[:,0]

pl = pv.Plotter()
pl.add_mesh(mesh, scalar_bar_args={"vertical": False})

# -----------------------------------------------------------------------------
# GUI
# -----------------------------------------------------------------------------

with SinglePageLayout(server) as layout:

     with layout.content:
        with vuetify3.VContainer(
            fluid=True,
            classes="pa-0 fill-height",
            style="position: relative;"
        ):
            view = plotter_ui(pl)

server.start()
```
I set the scalar bar to be horizontal. In remote rendering, it's OK, but in local rendering, it's still vertical.
Remote Rendering:
![image](https://github.com/Kitware/trame/assets/114913030/571fc8ba-2c99-41a0-a37d-d0bf4c43f78d)
Local Rendering:
![image](https://github.com/Kitware/trame/assets/114913030/9f473fe5-9cde-45b1-93dc-9bcacdf931ee)

If we change `scalar_bar_args` to {"vertical": True}, in remote rendering, the scalar bar will change to be vertical, but for local rendering, nothing changes.
Remote Rendering:
![image](https://github.com/Kitware/trame/assets/114913030/a0f389be-b965-4c1d-b94f-3900afea95aa)
Local Rendering:
![image](https://github.com/Kitware/trame/assets/114913030/c1600a36-7873-43e3-b2f9-db9673d5bf6a)

I don't know whether it's a bug, or I did something wrong.

**Comment by jourdain** [comment, marked as answer]
It is a limitation of vtk.js and how the VtkLocalView works.

**Reply by Jimmy-KL** [reply]
If I understand right, it's impossible to modify the scalar bar in LocalView mode, right? Or I have to define my own scalar bar in JS, right?

**Reply by jourdain** [reply]
Yes not easy to fix right now.

**Reply by jourdain** [reply]
Actually, maybe there is an option to set it horizontal in vtk.js. I just don't remember.

**Reply by Jimmy-KL** [reply]
OK, thanks.
